### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by nedrebo**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+jobs:
+- job: Windows
+  pool:
+    vmImage: vs2017-win2016
+  strategy:
+    matrix:
+      Python35:
+        python.version: '3.5'
```
Comment (trym-b) [inline azure-pipelines.yml]: Do we want f-strings in our code?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+jobs:
+- job: Windows
+  pool:
+    vmImage: vs2017-win2016
```
Comment (trym-b) [inline azure-pipelines.yml]: is there a list of combinations or something?

File: continuous-integration/windows/build.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import argparse
+import tempfile
+import sys
+import subprocess
+from pathlib import Path
+import requests
+
+
+def _get_options():
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: I like the _ in front of the functions, but I would have named this one: "_args()". You don't have to change it

File: continuous-integration/windows/build.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--root", required=True, help=f"The repository root")
+    parser.add_argument("--setup", action="store_true")
+    parser.add_argument("--build", action="store_true")
+    parser.add_argument("--test", action="store_true")
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: help is missing

File: continuous-integration/windows/build.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    _run_process(
+        ("echo", "TODO: ", "python", "-m", "pytest", root, "-c", rf"{root}\pytest.ini")
+    )
+
+
+def _main():
+    options = _get_options()
+
+    if options.setup:
+        _setup()
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: Invert this: use --skip-tests instead of --test

File: continuous-integration/windows/build.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+def _setup():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        zivid_installer_url = (
+            "https://www.zivid.com/hubfs/softwarefiles/ZividSetup_1.3.0+bb9ee328-10.exe"
+        )
+        print(f"Downloading {zivid_installer_url}")
+        zivid_installer = Path(temp_dir) / "ZividSetup.exe"
+        response = requests.get(zivid_installer_url)
+        zivid_installer.write_bytes(response.content)
+        print(f"Installing {zivid_installer}")
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: flush on all print statements?

File: continuous-integration/windows/build.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    except Exception as ex:
+        raise type(ex)(f"Process failed: '{' '.join(args)}'.") from ex
+    finally:
+        sys.stdout.flush()
+
+
+def _setup():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        zivid_installer_url = (
+            "https://www.zivid.com/hubfs/softwarefiles/ZividSetup_1.3.0+bb9ee328-10.exe"
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: this is not the first time we use zivid.com url, perhaps import it from one single module?

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      Python36:
+        python.version: '3.6'
+      Python37:
+        python.version: '3.7'
+  steps:
+  - script: pip install -r continuous-integration\requirements.txt"
+  - script: python continuous-integration\windows\build.py
+              --root %CD%
+              --setup --build --test
+    env: { 'Zivid_DIR': 'C:\Program Files\Zivid\CMake\Zivid', 'CXX': 'cl.exe', 'CC': 'cl.exe'}
```
Comment (trym-b) [inline azure-pipelines.yml]: I assume this is appending to PATH? Is clang on root level?

File: azure-pipelines.yml (hunk 1 above)
Comment (nedrebo) [reply azure-pipelines.yml]: Ah, lets see at least the testing of the package and the samples should work on Ubuntu-16.04(Python 3.5), that is why I included it. For now it is only building.

File: azure-pipelines.yml (hunk 2 above)
Comment (nedrebo) [reply azure-pipelines.yml]: What do you mean? vs2017-win2016 is the only image capable of building Python modules by default. We don't want setup hell.

File: continuous-integration/windows/build.py (hunk 7 above)
Comment (nedrebo) [reply continuous-integration/windows/build.py]: This is a fixed url, not something we generate and have control over

File: continuous-integration/windows/build.py (hunk 5 above)
Comment (nedrebo) [reply continuous-integration/windows/build.py]: I considered negating the options, but I did not find it better.

File: continuous-integration/windows/build.py (hunk 7 above)
Comment (trym-b) [reply continuous-integration/windows/build.py]: I mean https://www.zivid.com/blablabla -> f"{zivid_url}/blablabla"

File: continuous-integration/windows/build.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import sys
+import subprocess
+from pathlib import Path
+import requests
+
+
+def _options():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--root", required=True, help=f"The repository root")
```
Comment (trym-b) [inline continuous-integration/windows/build.py]: type=Path

**Pull request merged**